### PR TITLE
Item Protection

### PR DESCRIPTION
**Pull request by AzureAaron**
Ever felt like slot locking wasn't enough to protect your items? Ever accidentally almost tossed out that prized sword of yours? Or have you ever accidentally sold a precious item to the NPC (and maybe not gotten it back)?

Now meet `/skyblocker protectItem`!

When an item is protected it's safe from:
- Being dropped while holding it, regardless of the slot being locked
- Being dropped from picking it up in your inventory then clicking outside the screen
- Being dropped via hovering over it and pressing the drop key
- Being sold to the NPC
- Being inserted into the salvaging menu

Now you'll never have to worry about your valued items!

### CODE REVIEW

**Comment by Julienraptor01** [comment]
Can you trigger the action via a keybind ?
It would lock the item when hovering it or holding it in hand

File: src/main/java/me/xmrvizzy/skyblocker/mixin/HandledScreenMixin.java [hunk 1]
```diff
@@ -120,4 +136,53 @@ protected HandledScreenMixin(Text title) {
             }
         }
     }
+    
+    /**
+     * The naming of this method in yarn is half true, its mostly to handle slot/item interactions (which are mouse or keyboard clicks)
+     * 
+     * For example, using the drop key bind while hovering over an item will invoke this method to drop the players item
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/HandledScreenMixin.java:143]:
```suggestion
     * The naming of this method in yarn is half true, its mostly to handle slot/item interactions (which are mouse or keyboard clicks).
     * For example, using the drop key bind while hovering over an item will invoke this method to drop the players item.
```

File: src/main/java/me/xmrvizzy/skyblocker/mixin/HandledScreenMixin.java [hunk 2]
```diff
@@ -120,4 +136,53 @@ protected HandledScreenMixin(Text title) {
...
+            	
+                //Prevent salvaging
+                if (this.getTitle().getString().equals("Salvage Items")) {
+                    ItemStack stack = slot.getStack();
+
+                    if (ItemProtection.isItemProtected(stack)) ci.cancel();
+                }
+                
+                //Prevent selling to NPC shops
+                if (this.handler instanceof GenericContainerScreenHandler handler && handler.getRows() == 6) {
```
Comment (kevinthegreat1) [inline src/main/java/me/xmrvizzy/skyblocker/mixin/HandledScreenMixin.java:171]:
```suggestion
                if (this.client != null && this.handler instanceof GenericContainerScreenHandler genericContainerScreenHandler && genericContainerScreenHandler.getRows() == 6) {
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

**Comment by AzureAaron** [comment]
> Can you trigger the action via a keybind ?
> It would lock the item when hovering it or holding it in hand

This isn't really slot or item locking, intended to be applied for the long-term rather than short term

**Review by kevinthegreat1 — APPROVED** [review]

**Comment by kevinthegreat1** [comment]
Wait, maybe `ItemProtection` could go in the `item` package?

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**